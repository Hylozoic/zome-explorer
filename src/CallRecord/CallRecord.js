import React from 'react'
import JSONPretty from 'react-json-pretty';

const jsonTheme = {
  key: 'color:#80B21C;',
  string: 'color:#C455A6;',
  value: 'color:#C455A6;',
  boolean: 'color:#C455A6;'
}
export default function CallRecord ({ callRecord }) {
  const { path, params, result } = callRecord

return <div className='call-record'>
    <div className='call-record-path'>{path}</div>
    <div className='call-record-title'>Inputs</div>
    <JSONPretty data={params} theme={jsonTheme}/>
    <div className='call-record-title'>Outputs</div>
    <JSONPretty data={result} theme={jsonTheme} />    
  </div>
}