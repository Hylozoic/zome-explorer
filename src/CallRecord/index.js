import CallRecord from './CallRecord'
export default CallRecord