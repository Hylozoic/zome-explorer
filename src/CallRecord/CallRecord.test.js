import CallRecord from './CallRecord'
import { shallow } from 'enzyme'
import React from 'react'

it('matches the last snapshot', () => {
  const callRecord = {
    path: '/instance/zome/function',
    params: {
      a: "value",
      b: "different value"
    },
    result: {
      "ok": "yup"
    }
  }
  const wrapper = shallow(<CallRecord callRecord={callRecord} />)
  expect(wrapper).toMatchSnapshot()
})
