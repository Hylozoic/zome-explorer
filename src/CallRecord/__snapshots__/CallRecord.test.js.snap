// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the last snapshot 1`] = `
<div
  className="call-record"
>
  <div
    className="call-record-path"
  >
    /instance/zome/function
  </div>
  <div
    className="call-record-title"
  >
    Inputs
  </div>
  <JSONPretty
    data={
      Object {
        "a": "value",
        "b": "different value",
      }
    }
    json=""
    silent={true}
    space={2}
    theme={
      Object {
        "boolean": "color:#C455A6;",
        "key": "color:#80B21C;",
        "string": "color:#C455A6;",
        "value": "color:#C455A6;",
      }
    }
    themeClassName="__json-pretty__"
  />
  <div
    className="call-record-title"
  >
    Outputs
  </div>
  <JSONPretty
    data={
      Object {
        "ok": "yup",
      }
    }
    json=""
    silent={true}
    space={2}
    theme={
      Object {
        "boolean": "color:#C455A6;",
        "key": "color:#80B21C;",
        "string": "color:#C455A6;",
        "value": "color:#C455A6;",
      }
    }
    themeClassName="__json-pretty__"
  />
</div>
`;
